### PR TITLE
Ajay tripathy gpu csv

### PR DESCRIPTION
**Pull request by AjayTripathy**
## What does this PR change?
* Adds Custom Pricing for GPU support

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Allow nvidia.com/GPU support for on-prem deployments

## How was this PR tested?
* Unit tests

## Does this PR require changes to documentation?
* We should document adding CSV support for on-prem GPU deployments

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* Yes

### CODE REVIEW

File: pkg/cloud/csvprovider.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ type CSVProvider struct {
 	NodeMapField            string
 	PricingPV               map[string]*price
 	PVMapField              string
+	GPUClassPricing         map[string]*price
+	GPUMapFields            []string
```
Comment (michaelmdresser) [inline pkg/cloud/csvprovider.go]: What is `GPUMapFields`?

File: pkg/cloud/csvprovider.go [hunk 2]
```diff
@@ -202,30 +225,49 @@ func (k *csvKey) ID() string {
...
+	}
+
+	if node != nil {
+		if t := key.GPUType(); t != "" {
+			t = strings.ToLower(t)
+			count := key.GPUCount()
+			node.GPU = strconv.Itoa(count)
+			hourly := 0.0
+			if p, ok := c.GPUClassPricing[t]; ok {
+				hourly, _ = strconv.ParseFloat(p.MarketPriceHourly, 64)
```
Comment (michaelmdresser) [inline pkg/cloud/csvprovider.go]: Swallowing this error feels a little dangerous.

File: pkg/cloud/csvprovider.go [hunk 3]
```diff
@@ -202,30 +225,49 @@ func (k *csvKey) ID() string {
...
+			}
+			totalCost := hourly * float64(count)
+			node.GPUCost = fmt.Sprintf("%f", totalCost)
+			nc, _ := strconv.ParseFloat(node.Cost, 64)
```
Comment (michaelmdresser) [inline pkg/cloud/csvprovider.go]: Same here.

**Review by michaelmdresser — COMMENTED** [review]
Love the unit test!

File: pkg/cloud/csvprovider.go (hunk 1 above)
Comment (michaelmdresser) [reply pkg/cloud/csvprovider.go]: This is just for my own understanding!

Comment (AjayTripathy) [reply pkg/cloud/csvprovider.go]: This represents the possible fields in the node.labels that could correspond to a GPU. I can drop a comment.

**Review by michaelmdresser — APPROVED** [review]

**Comment by jessegoodier** [comment]
will we be merging this in nightly soon? happy to test it when ready.

**Comment by AjayTripathy** [comment]
@jessegoodier merged! Let's give it a test when you can!

**Comment by jessegoodier** [comment]
is this PR in the current nightly? `cost-model-nightly:OC-9b2d30f_KCM-77a689f`
If so, I might need some help.

**Comment by jessegoodier** [comment]
this is the cluster: https://kubecost.slack.com/archives/G01MN4R06MB/p1668783089270299

**Comment by michaelmdresser** [comment]
This PR is indeed in the current nightly. You can always check that image tag, which contains a shorthash of each repo's commit it was built from, against the repo commit history on `develop` to be sure: https://github.com/opencost/opencost/commits/develop

**Comment by jessegoodier** [comment]
It is working. Going to document now.

**Comment by brstuder** [comment]
Hey Jesse, can you rope me in on any docs-related work for this PR? Currently wondering where this info should best be added, or if it needs its own doc

**Comment by AjayTripathy** [comment]
@bstuder99 you may want to use the ping (cc @jessegoodier)

**Pull request merged**